qualtran/resource_counting/classify_bloqs.py

```diff
@@ -0,0 +1,88 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from collections import defaultdict
+from typing import Dict, Optional, Tuple, Union
+
+import sympy
+
+from qualtran import Bloq
+from qualtran.resource_counting.generalizers import (
+    ignore_alloc_free,
+    ignore_cliffords,
+    ignore_partition,
+    ignore_split_join,
+)
+from qualtran.resource_counting.t_counts_from_sigma import t_counts_from_sigma
+
+
+def _get_basic_bloq_classification() -> Dict[str, Tuple[str]]:
+    """High level classification of bloqs by the module name."""
+    bloq_classifier = {
+        'arithmetic': ('qualtran.bloqs.arithmetic',),
+        'rotations': ('qualtran.bloqs.rotations',),
+        'state_preparation': ('qualtran.bloqs.state_preparation',),
+        'data_loading': ('qualtran.bloqs.data_loading',),
+        'mcmt': ('qualtran.bloqs.mcmt',),
+        'multiplexers': ('qualtran.bloqs.multiplexers',),
+        'swaps': ('qualtran.bloqs.swap_network', 'qualtran.bloqs.basic_gates.swap'),
+        'reflection': ('qualtran.bloqs.reflection',),
+        'toffoli': ('qualtran.bloqs.basic_gates.toffoli',),
+        'tgate': ('qualtran.bloqs.basic_gates.t_gate',),
+    }
```

Comment:
Sorry for stringing you along, but what do you think about making the full name even more influential. Instead of a dict you can use a callable

```python
def classify(bloq_cls):
  mod_name = bloq_cls.__module__
  q, b, pkg, *_ = mod_name.split('.')
  if f'{q}.{b}' != 'qualtran.bloqs':
    return 'other'
  if bloq_cls == TGate:
    return 'tgate'
  return pkg
```

Reply:
I just pushed an updated version. I think what I have is a bit more flexible as it won't necessarily rely on the module structure. For example, what if I want to be more specific and classify 'data_loading' into 'qrom' and 'select_swap_qrom', which would require some further logic?

Reply:
Or for example, currently I'm combining swaps into one bucket. Granted the combination could be done using a second pass if you'd prefer?

Reply:
@mpharrigan WDYT: should it be a two stage thing with sub-classification happening after 'pkg' level classification?